### ISSUE TITLE
Bump specific v1.2.3 to major v2

### ISSUE THREAD

**Issue by ruudk** [dependabot/dependabot-core#7966]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

actions

### What you expected to see, versus what you actually saw

I have a GitHub action workflow that looks like:
```yaml
- uses: actions/checkout@v3.3.0
```

Dependabot comes and bumps it to 
```yaml
- uses: actions/checkout@v4
```

I use specific versions for a reason. I don't want to bump to yolo majors. Dependabot should bump to `@v4.0.0`.

**Comment by ruudk** [comment]
I found out why this happens. Out of my 11 actions/checkout, there was 1 acton that did `@v3`. Didn't know.
Would be great if I could configure Dependabot to always bump every implicit version to an explicit tagged one. 

So whenever a new developer accidentally uses `@v3` (as docs often tell you to do), Dependabot should fix them immediately to `@v3.0.0`.

**Comment by yeikel** [comment]
> I found out why this happens. Out of my 11 actions/checkout, there was 1 acton that did `@v3`. Didn't know. Would be great if I could configure Dependabot to always bump every implicit version to an explicit tagged one.
> 
> So whenever a new developer accidentally uses `@v3` (as docs often tell you to do), Dependabot should fix them immediately to `@v3.0.0`.

I agree that this is a bug, but I disagree that dependabot should make any such decision. 

I think that dependabot should instead follow the existing patterns in the project and if one action is using `v3` then the upgrade should be `v4`

**Comment by yeikel** [comment]
I think that dependabot should indeed follow the format and not make any decisions like this

> Would be great if I could configure Dependabot to always bump every implicit version to an explicit tagged one.

Given that this is a feature request, can you file a separate issue?

**Comment by ruudk** [comment]
In the situation of mixed formats (pinned vs no-pinned) it should always choose pinned. Currently, it chooses to convert everything to non-pinned. That's a far worse outcome.

**Comment by yeikel** [comment]
> In the situation of mixed formats (pinned vs no-pinned) it should always choose pinned. Currently, it chooses to convert everything to non-pinned. That's a far worse outcome.

I don't think it should make that decision. It should follow the format that the team is using 

In some situations, it can be desired to have that (i.e., don't pin internal actions but pin external) and dependabot shouldn't make that decision for users
